File: swift/Sources/AmenClient/Resources/Withdrawals.swift
public final class Withdrawals {
    let c: AmenClient; init(_ c: AmenClient) { self.c = c }
    public func create(bankAccountId: String, amount: String) async throws -> Withdrawal { try await c.request(.POST, "/withdrawals/", json: ["bank_account_id": bankAccountId, "amount": amount]) }
    public func get(_ n: String) async throws -> Withdrawal { try await c.request(.GET, "/withdrawals/\(n)") }
    public func list(page: Int? = nil, pageSize: Int? = nil, status: String? = nil) async throws -> Page<Withdrawal> { try await c.requestPage(.GET, "/withdrawals/", key: "withdrawals", params: ["page": page.map(String.init), "page_size": pageSize.map(String.init), "status": status]) }
}

File: flutter/amen_client/lib/src/resources/withdrawals.dart
import '../client.dart';
import '../models.dart';

class Withdrawals {
  final AmenClient _c;
  Withdrawals(this._c);
  Future<Withdrawal> create({required String bankAccountId, required String amount}) async =>
      Withdrawal((await _c.request('POST', '/withdrawals/', json: {'bank_account_id': bankAccountId, 'amount': amount}) as Map).cast());
  Future<Withdrawal> get(String n) async => Withdrawal((await _c.request('GET', '/withdrawals/$n') as Map).cast());
  Future<Page<Withdrawal>> list({int? page, int? pageSize, String? status}) async =>
      Page.from((await _c.request('GET', '/withdrawals/', params: {'page': page?.toString(), 'page_size': pageSize?.toString(), 'status': status}) as Map?)?.cast(), 'withdrawals', Withdrawal.new);
}
